src/main/kotlin/uk/gov/gdx/datashare/queue/HmppsQueueResourceAsync.kt

```diff
@@ -0,0 +1,54 @@
+package uk.gov.gdx.datashare.queue
+
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+
+/*
+ * Warning: This is a duplicate of HmppsQueueResource but with suspend functions. Therefore, this class should be kept in sync with the non-async class.
+ *
+ * This class *should* just be a wrapper around HmppsQueueResource but that sometimes works and sometimes doesn't! So we're keeping the duplication for now.
+ */
+@RestController
+@RequestMapping("/queue-admin")
+class HmppsQueueResourceAsync(private val hmppsQueueService: HmppsQueueService) {
```

Comment:
should this be a controller?

Reply:
Yes

Reply:
I'm tempted to outright remove this anyway, but might leave that as an exercise for later